File: telerik_ui_for_winforms_courseware_chm/Courseware/TPF/CS/FisheyeBar/WindowsFormsApplication1/RadStackItemsPanel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing.Design;
using System.Text;
using Telerik.WinControls;
using Telerik.WinControls.Layouts;

namespace Telerik.QuickStart.WinControls
{
    public class RadStackItemsPanel : RadControl
    {
        private RadStackItemsPanelElement stackPanelElement;

        public RadStackItemsPanel()
        {
            this.UseNewLayoutSystem = true;
        }

        protected override void CreateChildItems(RadElement parent)
        {
            base.CreateChildItems(parent);

            stackPanelElement = new RadStackItemsPanelElement();
            stackPanelElement.SerializeElement = false;

            parent.Children.Add(stackPanelElement);
        }

        [RadEditItemsAction]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public RadItemOwnerCollection Items
        {
            get
            {
                return this.PanelElement.Items;
            }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public RadStackItemsPanelElement PanelElement
        {
            get
            {
                return this.stackPanelElement;
            }
        }
    }
}

File: telerik_ui_for_winforms_courseware_chm/Courseware/TPF/VB/FisheyeBar/FisheyeBar/RadStackItemsPanel.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing.Design
Imports System.Text
Imports Telerik.WinControls
Imports Telerik.WinControls.Layouts

Public Class RadStackItemsPanel
    Inherits RadControl
    Private stackPanelElement As RadStackItemsPanelElement

    Public Sub New()
        Me.UseNewLayoutSystem = True
    End Sub

    Protected Overrides Sub CreateChildItems(ByVal parent As RadElement)
        MyBase.CreateChildItems(parent)

        stackPanelElement = New RadStackItemsPanelElement()
        stackPanelElement.SerializeElement = False

        parent.Children.Add(stackPanelElement)
    End Sub

    <RadEditItemsAction()> _
     <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
     Public ReadOnly Property Items() As RadItemOwnerCollection
        Get
            Return Me.PanelElement.Items
        End Get
    End Property

    <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
 Public ReadOnly Property PanelElement() As RadStackItemsPanelElement
        Get
            Return Me.stackPanelElement
        End Get
    End Property

End Class